File: Benchmarks_FSharp/Core/Types/GameTime.fs
namespace Benchmarks.Core.Types

open System

type GameTime(totalGameTime: TimeSpan, elapsedGameTime: TimeSpan) =
    member val TotalGameTime = totalGameTime with get, set
    member val ElapsedGameTime = elapsedGameTime with get, set
    new() = GameTime(TimeSpan.Zero, TimeSpan.Zero)

File: Benchmarks/Core/Types/GameTime.cs
namespace Benchmarks.Core.Types
{
    public class GameTime
    {
        public TimeSpan TotalGameTime { get; set; }
        public TimeSpan ElapsedGameTime { get; set; }

        public GameTime() : this(TimeSpan.Zero, TimeSpan.Zero) { }

        public GameTime(TimeSpan totalGameTime, TimeSpan elapsedGameTime)
        {
            TotalGameTime = totalGameTime;
            ElapsedGameTime = elapsedGameTime;
        }
    }
}
